src/pages/[componentAccountId]/widget/[componentName].tsx

```diff
@@ -12,7 +16,76 @@ import { useCurrentComponentStore } from '@/stores/current-component';
 import { privacyDomainName, termsDomainName } from '@/utils/config';
 import type { NextPageWithLayout } from '@/utils/types';
 
-const ViewComponentPage: NextPageWithLayout = () => {
+type ComponentMetaPreview = {
+  title: string;
+  description: string;
+  imageUrl: string | null;
+};
+
+type ComponentPayload = Record<string, { widget: Record<string, { '': string; metadata: ComponentMetadata }> }>;
+type ComponentMetadata = {
+  name: string;
+  description: string;
+  linktree: {
+    website: string;
+  };
+  image: ImageData;
+  tags: Record<string, string>;
+};
+
+type ImageData = {
+  ipfs_cid?: string;
+};
+
+function returnImageUrl(data: ImageData | undefined) {
+  if (data?.ipfs_cid) {
+    return `https://i.near.social/large/https://ipfs.near.social/ipfs/${data.ipfs_cid}`;
+  }
+  return null;
+}
+
+async function fetchPreviewData(accountId: string, componentName: string): Promise<ComponentMetaPreview | null> {
+  const response = await fetch(`https://api.near.social/get?keys=${accountId}/widget/${componentName}/**`);
```

Comment:
@calebjacob I think we could use a try/catch here to not block page load in the event when this API is degraded.

Reply:
+1

Reply:
Thanks for pointing that out! Great idea. Pushed a commit to add the `try/catch` wrapper here.